packages/@biomejs/tailwindcss-config-analyzer/src/generate-tailwind-preset.ts

```diff
@@ -0,0 +1,87 @@
+import { introspectTailwindConfig } from "./introspect.js";
+import { SortConfig, sortConfigFromSpec } from "./sort-config.js";
+import { readPackageUp } from "read-package-up";
+import path from "node:path";
+
+const ROOT_PACKAGE_NAME = "@biomejs/monorepo";
+const OUTPUT_PATH =
+	"crates/biome_js_analyze/src/semantic_analyzers/nursery/use_sorted_classes/tailwind_preset.rs";
+
+const EXCLUDED_LAYERS = ["defaults", "base"];
+const LAYER_ORDER = ["components", "utilities"];
+
+const LINE_LIMIT = 100;
+const INDENT = "    ";
+const FILE_HEADER = `//! DO NOT EDIT MANUALLY - this file is autogenerated from the default Tailwind CSS config.
+//! To update, run \`pnpm gen:tailwind-preset\` from \`packages/@biomejs/tailwindcss-config-analyzer\`.
+
+use super::sort_config::UtilityLayer;
+`;
+
+function generateLayer({ layer, classes }: SortConfig["utilities"][number]) {
+	const header = `const ${layer.toUpperCase()}_LAYER_CLASSES: [&str; ${classes.length}] = [`;
+
+	// try single line
+	const singleLine = `${header}${classes.map((c) => `"${c}"`).join(", ")}];`;
+	if (singleLine.length < LINE_LIMIT) return `${singleLine}\n`;
+
+	// if single line is too long, do multi-line
+	return `${header}\n${classes.map((c) => `${INDENT}"${c}"`).join(",\n")},\n];\n`;
```

Comment:
This line can be a single reduce call

Reply:
I find it more readable like this. I don't think it's worth bikeshedding over since this is just an internal codegen script that is only used on-demand (and probably only necessary when there's a new tailwind version that adds built-in utilities or variants).